src/ripple/nodestore/impl/BatchWriter.cpp

```diff
@@ -39,7 +39,12 @@ BatchWriter::~BatchWriter ()
 void
 BatchWriter::store (std::shared_ptr<NodeObject> const& object)
 {
-    std::lock_guard<decltype(mWriteMutex)> sl (mWriteMutex);
+    std::unique_lock<decltype(mWriteMutex)> sl (mWriteMutex);
+
+    // If the batch has reached its limit, we wait
+    // until the batch writer is finished
+    while (mWriteSet.size() >= batchWriteLimitSize)
```

Comment:
Nit: I believe this should be `>`.

This can probably be an `if` statement as I doubt we will ever have greater than `batchWriteLimitSize` threads waiting.

Reply:
If the size is equal to the limit, we want to wait so that we don't exceed the limit. It has to be a while because spurious wakeups are a real thing and we don't want to risk randomly exceeding the limit.